### PR TITLE
style: update golangci config and fix lint errors

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

This pull request updates the lint configuration and fixes lint errors and warnings. This pull request also updates workflows to ensure each module is checked if changes are made within the given module and skips checks if no changes are made.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: types/module/client/grpc_gateway/module.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package grpc_gateway
+package grpcgateway
```
Comment (ryanchristo) [inline types/module/client/grpc_gateway/module.go:1]: breaking change

File: x/ecocredit/basket/date_criteria.go [hunk 2]
```diff
@@ -7,8 +7,8 @@ import (
 	"github.com/regen-network/regen-ledger/types"
 )
 
-// ToApi converts to pulsar based data structure
-func (d *DateCriteria) ToApi() *api.DateCriteria {
+// ToAPI converts to pulsar based data structure
+func (d *DateCriteria) ToAPI() *api.DateCriteria {
```
Comment (ryanchristo) [inline x/ecocredit/basket/date_criteria.go:11]: breaking change

File: x/ecocredit/client/query.go [hunk 3]
```diff
@@ -473,8 +473,8 @@ func QueryParamsCmd() *cobra.Command {
 	})
 }
 
-// QueryProjectsByReferenceIdCmd returns command that retrieves list of projects by reference id with pagination.
-func QueryProjectsByReferenceIdCmd() *cobra.Command {
+// QueryProjectsByReferenceIDCmd returns command that retrieves list of projects by reference id with pagination.
+func QueryProjectsByReferenceIDCmd() *cobra.Command {
```
Comment (ryanchristo) [inline x/ecocredit/client/query.go:477]: breaking change

File: x/ecocredit/client/tx.go [hunk 4]
```diff
@@ -23,7 +23,7 @@ import (
 const (
 	FlagAddIssuers             string = "add-issuers"
 	FlagRemoveIssuers          string = "remove-issuers"
-	FlagReferenceId            string = "reference-id"
+	FlagReferenceID            string = "reference-id"
```
Comment (ryanchristo) [inline x/ecocredit/client/tx.go:26]: breaking change

File: x/ecocredit/core/msg_create_project.go [hunk 5]
```diff
@@ -8,7 +8,7 @@ import (
 	"github.com/regen-network/regen-ledger/x/ecocredit"
 )
 
-const MaxReferenceIdLength = 32
+const MaxReferenceIDLength = 32
```
Comment (ryanchristo) [inline x/ecocredit/core/msg_create_project.go:11]: breaking change

File: x/ecocredit/core/utils.go [hunk 6]
```diff
@@ -35,18 +33,18 @@ const (
 )
 
 var (
-	RegexClassId      = `[A-Z]{1,3}[0-9]{2,}`
-	RegexProjectId    = fmt.Sprintf(`%s-[A-Z0-9]{2,}`, RegexClassId)
-	RegexBatchDenom   = fmt.Sprintf(`%s-[0-9]{8}-[0-9]{8}-[0-9]{3,}`, RegexProjectId)
+	RegexClassID      = `[A-Z]{1,3}[0-9]{2,}`
+	RegexProjectID    = fmt.Sprintf(`%s-[A-Z0-9]{2,}`, RegexClassID)
```
Comment (ryanchristo) [inline x/ecocredit/core/utils.go:37]: breaking change

File: x/ecocredit/core/utils.go [hunk 7]
```diff
@@ -58,24 +56,24 @@ var (
 // least three digits
 //
 // e.g. C01
-func FormatClassId(creditTypeAbbreviation string, classSeqNo uint64) string {
+func FormatClassID(creditTypeAbbreviation string, classSeqNo uint64) string {
```
Comment (ryanchristo) [inline x/ecocredit/core/utils.go:59]: breaking change

File: x/ecocredit/core/utils.go [hunk 8]
```diff
@@ -58,24 +56,24 @@ var (
...
 //
-// <class-id> is the unique identifier of the credit class (see FormatClassId)
+// <class-id> is the unique identifier of the credit class (see FormatClassID)
 // <project-sequence> is the sequence number of the project, padded to at least
 // three digits
 //
 // e.g. C01-001
-func FormatProjectId(classId string, projectSeqNo uint64) string {
-	return fmt.Sprintf("%s-%03d", classId, projectSeqNo)
+func FormatProjectID(classID string, projectSeqNo uint64) string {
```
Comment (ryanchristo) [inline x/ecocredit/core/utils.go:75]: breaking change

File: x/ecocredit/core/utils.go [hunk 9]
```diff
@@ -122,28 +120,28 @@ func ValidateCreditTypeAbbreviation(abbr string) error {
 	return nil
 }
 
-// ValidateClassId validates a class ID conforms to the format described in
-// FormatClassId. The return is nil if the ID is valid.
-func ValidateClassId(classId string) error {
-	if classId == "" {
+// ValidateClassID validates a class ID conforms to the format described in
+// FormatClassID. The return is nil if the ID is valid.
+func ValidateClassID(classID string) error {
```
Comment (ryanchristo) [inline x/ecocredit/core/utils.go:125]: breaking change

File: x/ecocredit/core/utils.go [hunk 10]
```diff
@@ -122,28 +120,28 @@ func ValidateCreditTypeAbbreviation(abbr string) error {
...
 	return nil
 }
 
-// ValidateProjectId validates a project ID conforms to the format described
-// in FormatProjectId. The return is nil if the ID is valid.
-func ValidateProjectId(projectId string) error {
-	if projectId == "" {
+// ValidateProjectID validates a project ID conforms to the format described
+// in FormatProjectID. The return is nil if the ID is valid.
+func ValidateProjectID(projectID string) error {
```
Comment (ryanchristo) [inline x/ecocredit/core/utils.go:138]: breaking change

File: x/ecocredit/core/utils.go [hunk 11]
```diff
@@ -178,10 +176,10 @@ func ValidateJurisdiction(jurisdiction string) error {
 	return nil
 }
 
-// GetClassIdFromProjectId returns the credit class ID in a project ID.
-func GetClassIdFromProjectId(projectId string) string {
+// GetClassIDFromProjectID returns the credit class ID in a project ID.
+func GetClassIDFromProjectID(projectID string) string {
```
Comment (ryanchristo) [inline x/ecocredit/core/utils.go:180]: breaking change

File: x/ecocredit/core/utils.go [hunk 12]
```diff
@@ -191,8 +189,8 @@ func GetClassIdFromProjectId(projectId string) string {
 	return s.String()
 }
 
-// GetClassIdFromBatchDenom returns the credit class ID in a batch denom.
-func GetClassIdFromBatchDenom(denom string) string {
+// GetClassIDFromBatchDenom returns the credit class ID in a batch denom.
+func GetClassIDFromBatchDenom(denom string) string {
```
Comment (ryanchristo) [inline x/ecocredit/core/utils.go:193]: breaking change

File: x/ecocredit/core/utils.go [hunk 13]
```diff
@@ -204,8 +202,8 @@ func GetClassIdFromBatchDenom(denom string) string {
 	return s.String()
 }
 
-// GetProjectIdFromBatchDenom returns the credit project ID in a batch denom.
-func GetProjectIdFromBatchDenom(denom string) string {
+// GetProjectIDFromBatchDenom returns the credit project ID in a batch denom.
+func GetProjectIDFromBatchDenom(denom string) string {
```
Comment (ryanchristo) [inline x/ecocredit/core/utils.go:206]: breaking change

File: x/ecocredit/core/utils.go [hunk 14]
```diff
@@ -221,10 +219,10 @@ func GetProjectIdFromBatchDenom(denom string) string {
 	return s.String()
 }
 
-// GetCreditTypeAbbrevFromClassId returns the credit type abbreviation in a credit class id
-func GetCreditTypeAbbrevFromClassId(classId string) string {
+// GetCreditTypeAbbrevFromClassID returns the credit type abbreviation in a credit class id
+func GetCreditTypeAbbrevFromClassID(classID string) string {
```
Comment (ryanchristo) [inline x/ecocredit/core/utils.go:223]: breaking change

File: x/ecocredit/server/basket/invariants.go [hunk 15]
```diff
@@ -25,16 +25,16 @@ func (k Keeper) basketSupplyInvariant() sdk.Invariant {
...
 }
 
 type bankSupplyStore interface {
 	GetSupply(ctx sdk.Context, denom string) sdk.Coin
 }
 
-// BasketSupplyInvariant cross check the balance of baskets and bank
-func BasketSupplyInvariant(ctx sdk.Context, store api.BasketTable, bank bankSupplyStore, basketBalances map[uint64]math.Dec) (string, bool) {
+// SupplyInvariant cross check the balance of baskets and bank
+func SupplyInvariant(ctx sdk.Context, store api.BasketTable, bank bankSupplyStore, basketBalances map[uint64]math.Dec) (string, bool) {
```
Comment (ryanchristo) [inline x/ecocredit/server/basket/invariants.go:37]: breaking change

**Review by technicallyty — APPROVED** [review]
LGTM

**Pull request merged**